examples/custom_metrics/model.py

```diff
@@ -0,0 +1,174 @@
+        Batching) used, `requests` may contain multiple requests. Every
+        Python model, must create one pb_utils.InferenceResponse for every
+        pb_utils.InferenceRequest in `requests`. If there is an error, you can
+        set the error argument when creating a pb_utils.InferenceResponse
+
+        Parameters
+        ----------
+        requests : list
+          A list of pb_utils.InferenceRequest
+
+        Returns
+        -------
+        list
+          A list of pb_utils.InferenceResponse. The length of this list must
+          be the same as `requests`
+        """
+
+        responses = []
+
+        # Record the start time of processing the requests
+        start_ns = time.time_ns()
+        # Every Python backend must iterate over everyone of the requests
+        # and create a pb_utils.InferenceResponse for each of them.
+        for request in requests:
+            # Get INPUT0
+            in_0 = pb_utils.get_input_tensor_by_name(request, "INPUT0")
+            # Get INPUT1
+            in_1 = pb_utils.get_input_tensor_by_name(request, "INPUT1")
+
+            out_0, out_1 = (in_0.as_numpy() + in_1.as_numpy(),
+                            in_0.as_numpy() - in_1.as_numpy())
+
+            # Create output tensors. You need pb_utils.Tensor
+            # objects to create pb_utils.InferenceResponse.
+            out_tensor_0 = pb_utils.Tensor("OUTPUT0",
+                                           out_0.astype(self.out0_dtype))
+            out_tensor_1 = pb_utils.Tensor("OUTPUT1",
+                                           out_1.astype(self.out1_dtype))
+
+            # Create InferenceResponse. You can set an error here in case
+            # there was a problem with handling this inference request.
+            # Below is an example of how you can set errors in inference
+            # response:
+            #
+            # pb_utils.InferenceResponse(
+            #    output_tensors=..., TritonError("An error occured"))
+            inference_response = pb_utils.InferenceResponse(
+                output_tensors=[out_tensor_0, out_tensor_1])
+            responses.append(inference_response)
+
+        # Record the end time of processing the requests
+        end_ns = time.time_ns()
+
+        # Update metric to track cumulative requests processing latency.
+        # There are three operations you can do with the Metric object:
+        #   - Metric.increment(value): Increment the value of the metric by
+        #       the given value. The type of the value is double. The 'COUNTER'
+        #       kind does not support negative value.
+        #   - Metric.set(value): Set the value of the metric to the given
+        #       value. This operation is only supported in 'GAUGE' kind. The
+        #       type of the value is double.
+        #   - Metric.value(): Get the current value of the metric.
+        self.metric.increment(end_ns - start_ns)
```

Comment:
It might be a good idea to use the Logger in this example as well to test any potential issues with metrics/logs sharing the same queue.

Reply:
Good idea. I think adding the Logger in the testing would be better. commit: https://github.com/triton-inference-server/server/pull/5669/commits/79a5e9adc7af3ef3fe714034220cccc0d9a14965.